Replace Ctx.CustomObjects type with thread safe and secure ObjectMap API

zekroTJA opened the pull request:
Hey! 👋

I stumbled across this project a while ago and currently, I'm thinking about using it in the upstanding rework of yuri, so I've decided to try to give some improvements to this package, if wanted. ^^

First of all, using a raw `map[string]interface{}` as `Ctx.CustomObjects` might bring some nasty issues. Concurrently reading on maps in go is fine until you read/write on maps in parallel, which will lead to a panic.
Here you can read more about this with some handy examples:
https://qvault.io/2020/03/19/golang-mutexes-what-is-rwmutex-for

So i've simply wrapped the map with a `sync.RWMutex` which blocks access to the map while writing to it and also providing some nice API endpoints for accessing the CustomObjects field.

Also, I've simplified the inline object creation in the embed arrays of `helpCommand.go`.

Additionally, I've put the example from the README into the example directory so that people can test around with the package using the provided example.

zekroTJA committed:
replace CustomObject map with thread safe wrapper API

zekroTJA committed:
simplify inline object creation in helpCommand

zekroTJA committed:
add Delete function to ObjectMap

lus reviewed (changes requested):
Hey^^

First of all thanks for your efford to contribute to this project. I looked over the changes and the only thing I noticed are the key names in the `ObjectsMap` structure. These aren't self-explaining, which would be a great addition in my eyes. Sure, this is only a personal opinion, but I would like to keep this repository as uniform as possible. Would be nice if you could change them to something like `mutex` and `map` ;)

Thank you!

zekroTJA committed:
fix names of ObjectMap; bump dependency timedmap to 1.1.2

zekroTJA commented:
Okay, I've changed `m` to `innerMap` and `mx` to `mutex` for more clarity.
Also, I've updated the version of timedmap which got also an [update](https://github.com/zekroTJA/timedmap/releases/tag/1.1.2) this morning. ;)

lus commented:
Alright  seems good. Thank you!
